### DISCUSSION TITLE
Question about delayed loading of completion scripts

### DISCUSSION THREAD

**Discussion by tshu-w** [marlonrichert/zsh-snap#146] [Q&A] [answered]
Hi, I'm using [shtab](https://github.com/iterative/shtab#library-usage) to generate Python CLI applications completion scripts by `run -s zsh` and the generated scripts look like [this](https://github.com/tshu-w/deep-learning-project-template/blob/master/_run). I want to use znap lazy loading the generation of completion scripts as the cli  applications is project specific and I use direnv to add `$PWD` to `$PATH` when I enter the project directory. I tried the following configs but got `command not found: _run`. Thank you in advance.
```zsh
znap function _run run "znap fpath _run 'run -s zsh'"
compdef _run run
```

**Comment by marlonrichert** [comment, marked as answer]
Just run `znap fpath _run 'run -s zsh'` once on the command line to install the completion function. That's it. Run it again when you want to update the completion function. No need for the other calls.

**Reply by tshu-w** [reply]
Thank you reply. My question is after `znap fapth _run 'run -s zsh'` I need to restart zsh first to avoid `command not found: _run`. Is there any way to avoid to restart zsh. Does zsh register all the files in fpath at start-up?

**Reply by marlonrichert** [reply]
`zsh` does not do so by itself. Rather, when `compinit` is called, it registers all completion functions in the `$fpath`. However, invoking `compinit` more than once in the same shell session can lead to unexpected behavior. So, no, there's no way to avoid restarting `zsh` in this scenario.

**Reply by tshu-w** [reply]
Got it. Thank you for your reply!
